### DISCUSSION TITLE
Need some help in understanding - npm install failure

### DISCUSSION THREAD

**Discussion by 311chaos** [renovatebot/renovate#13990] [Request Help]
Hi all,

I am using the GItHub app to configure Renovate. When a PR is opened I get failures on the `npm install` step. Specifically:

```
npm ERR! code EBADPLATFORM
npm ERR! notsup Unsupported platform for fsevents@2.2.1
```

I am using a lock file in my project with a config of:

```
{
  "commitMessageAction": "[patch] - ",
  "dependencyDashboardApproval": true,
  "enabledManagers": ["npm"],
  "extends": ["config:base", ":dependencyDashboard"],
  "labels": ["dependencies"],
  "major": {
    "dependencyDashboardApproval": true
  },
  "npm": {
    "stabilityDays": 3
  },
  "packageRules": [
    {
      "matchDepTypes": ["devDependencies"],
      "matchUpdateTypes": ["patch"],
      "groupName": "devDependencies [patch]"
    },
    {
      "matchDepTypes": ["dependencies"],
      "matchUpdateTypes": ["patch"],
      "groupName": "Dependencies [patch]"
    }
  ],
  "prConcurrentLimit": 4,
  "prCreation": "not-pending"
}
```

Is there a way I can customize the `npm install` command to ignore optional dependencies? Or what other options should I be looking at within the config that might be of assistance for me? Thank you in advance to anyone who reads this far along! ;)

**Comment by rarkins** [comment]
npm should normally ignore optional dependencies anyway. This SO thread indicates that you might have something wrong with your lock file? https://stackoverflow.com/questions/49475492/npm-install-error-code-ebadplatform
